src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -1422,3 +1422,53 @@ function random_orientation_dag(
     end
     return g2
 end
+
+"""
+    bernoulli_graph(Λ)
+
+Given the symmetric matrix ``\\Lambda \\in [0,1]^{n \\times n}``, return a Bernoulli graph with ``n`` vertices. Each edge ``(i,j)`` exists with probability ``\\operatorname{Bernoulli}(\\Lambda[i,j])``.
+"""
+function bernoulli_graph(
+    Λ::Matrix{Float64};
+    rng::Union{Nothing,AbstractRNG}=nothing,
+    seed::Union{Nothing,Integer}=nothing,
+)
+    @assert size(Λ)[1] == size(Λ)[2] "The probability matrix must be a square matrix!"
+    n = size(Λ)[1]
+    rng = rng_from_rng_or_seed(rng, seed)
+    A = SimpleGraph(n)
+    for j in 1:n
+        for i in (j + 1):n
```

Comment:
So no self-loops allowed? In that case we should document this, or add a flag that specifies if we allow self-loops.